plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Entity,
+  RELATION_PROVIDES_API,
+  RELATION_PART_OF,
+  serializeEntityRef,
+} from '@backstage/catalog-model';
+import {
+  catalogApiRef,
+  getEntityRelations,
+} from '@backstage/plugin-catalog-react';
+import {
+  DependencyGraph,
+  DependencyGraphTypes,
+  InfoCard,
+  Progress,
+  useApi,
+} from '@backstage/core';
+import { Box, Typography } from '@material-ui/core';
+import { Alert } from '@material-ui/lab';
+import ZoomOutMap from '@material-ui/icons/ZoomOutMap';
+import React from 'react';
+import { useAsync } from 'react-use';
+
+type SystemDiagramProps = {
+  entity: Entity;
```

Comment:
Last piece of the migration is to remove this prop and grab the entity via `useEntity` from `@backstage/catalog-plugin-react` instead!